### PR TITLE
using vcr and a custom click vcr for tests

### PR DESCRIPTION
**Pull request by sbneto**
Migrating tests from `requests` to `vcr` and a custom implementation of `click vcr` that records the output of `click` commands. Used it in one failing test, but will migrate all tests over time. Will also test against `e2e` instead of stage.

### CODE REVIEW

**Review by alanjds — APPROVED** [review]
That is interesting. Is this VCR with Click capabilities available on PyPI or some Github repo?

**Comment by sbneto** [comment]
Nope, just some custom code I wrote in the base test class.

File: tests/vcr/test_download_cat.click [hunk 1]
```diff
@@ -0,0 +1 @@
+result: X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*
```
Comment (alanjds) [inline tests/vcr/test_download_cat.click:1]: My anti-virus is not gonna like this...

If possible (as a vcr option maybe), consider to use base64 or gzip encoding on this one.

**Review by alanjds — APPROVED** [review]

File: tests/vcr/test_download_cat.click (hunk 1 above)
Comment (sbneto) [reply tests/vcr/test_download_cat.click:1]: This is only in the tests, it is not distributed with the package.

Comment (mrtizmoatwork) [reply tests/vcr/test_download_cat.click:1]: generally, we should keep this as a base64 encoded string and then decode it when we need to do matching.

**Pull request merged**